packages/core/src/mappings/TextInput.ts

```diff
@@ -4,12 +4,391 @@ import {
+      "number-pad",
+      "name-phone-pad",
+      "decimal-pad",
+      "twitter",
+      "web-search",
+      "visible-password",
+    ],
+    formType: FORM_TYPES.flatArray,
+    propType: PROP_TYPES.STRING,
+  },
+  maxLength: {
+    group: GROUPS.basic,
+    label: "Max Length",
+    description: "Limits the input to a set number of characters.",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    min: 0,
+    step: 1,
+    precision: 0,
+    formType: FORM_TYPES.number,
+    propType: PROP_TYPES.NUMBER,
+  },
+  placeholder: {
+    group: GROUPS.data,
+    label: "Placeholder Text",
+    description: "The text that is shown on load when no value is available.",
+    editable: true,
+    required: false,
+    defaultValue: "Enter a value...",
+    formType: FORM_TYPES.string,
+    propType: PROP_TYPES.STRING,
+  },
+  placeholderTextColor: {
+    group: GROUPS.basic,
+    label: "Placeholder Text Color",
+    description: "The color of the placeholder text.",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    formType: FORM_TYPES.color,
+    propType: PROP_TYPES.STRING,
+  },
+  returnKeyLabel: {
+    group: GROUPS.advanced,
+    label: "Return Key Label",
+    description:
+      "(Android Only) Sets the label on the return key (use this instead of rewturnKeyType)",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    formType: FORM_TYPES.string,
+    propType: PROP_TYPES.STRING,
+  },
+  returnKeyType: {
+    group: GROUPS.advanced,
+    label: "Return Key Type",
+    description: "Determines how the return key should look like",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    options: [
+      "done",
+      "go",
+      "next",
+      "search",
+      "send",
+      "none",
+      "previous",
+      "default",
+      "emergency-call",
+      "google",
+      "join",
+      "route",
+      "yahoo",
+    ],
+    formType: FORM_TYPES.flatArray,
+    propType: PROP_TYPES.STRING,
+  },
+  secureTextEntry: {
+    group: GROUPS.basic,
+    label: "Password Input?",
+    description:
+      "Hides the characters with a *, useful for passwords and other sensitive information.",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    formType: FORM_TYPES.boolean,
+    propType: PROP_TYPES.BOOLEAN,
+  },
+  selectionColor: {
+    group: GROUPS.advanced,
+    label: "Selection Color",
+    description: "Color of the highlighted portion when selecting.",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    formType: FORM_TYPES.color,
+    propType: PROP_TYPES.STRING,
+  },
+  selectTextOnFocus: {
+    group: GROUPS.advanced,
+    label: "Select Text on Focus",
+    description:
+      "If true, all the text will automatically be selected on focus",
+    editable: true,
+    required: false,
+    defaultValue: null,
+    formType: FORM_TYPES.boolean,
+    propType: PROP_TYPES.BOOLEAN,
+  },
+};
+
+const NUMBER_INPUT_PROPS = {
```

Comment:
Not to pile more work on you, but can we move this and the SEED_DATA for the NumberInput component to a new file `NumberInput.ts`?

Reply:
Sure, I was planning to do it in a subsequent PR actually, but now's fine

Reply:
Done